test/cmd/newapp.sh

```diff
@@ -53,14 +53,25 @@ os::cmd::expect_success 'oc delete is myruby'
 os::cmd::expect_failure_and_text 'oc new-app https://github.com/openshift/nodejs-ex --strategy=docker' 'No Dockerfile was found'
 
 # repo related error message validation
-os::cmd::expect_failure_and_text 'oc new-app mysql-persisten mysql' 'mysql-persisten as a local directory'
-os::cmd::expect_failure_and_text 'oc new-app mysql-persisten mysql' 'mysql as a local directory'
-os::cmd::expect_failure_and_text 'oc new-app --strategy=docker https://192.30.253.113/openshift/ruby-hello-world.git' 'as a Git repository URL:  '
-os::cmd::expect_failure_and_text 'oc new-app https://www.google.com/openshift/nodejs-e' 'as a Git repository URL:  '
-os::cmd::expect_failure_and_text 'oc new-app https://examplegit.com/openshift/nodejs-e' 'as a Git repository URL:  '
-os::cmd::expect_failure_and_text 'oc new-build --strategy=docker https://192.30.253.113/openshift/ruby-hello-world.git' 'as a Git repository URL:  '
-os::cmd::expect_failure_and_text 'oc new-build https://www.google.com/openshift/nodejs-e' 'as a Git repository URL:  '
-os::cmd::expect_failure_and_text 'oc new-build https://examplegit.com/openshift/nodejs-e' 'as a Git repository URL:  '
+# oc new-app mysql-persisten mysql reacts differently when run after oc cluster up; with cluster up, it
+# finds the mysql-persistent template in the openshift namespace and cites a partial match, but in this test
+# that namespace/template combo does not exist, and so falls back to the generic not match / unable to locate resource
+#os::cmd::expect_failure_and_text 'oc new-app mysql-persisten mysql' 'only a partial match was found for'
```

Comment:
how about changing it to "oc new-app mysq" which should partial match against the mysql imagestream?

Reply:
Similar problem unfortunately

When trying that after `oc cluster up`, `oc new-app mysq` reports:

```
error: multiple images or templates matched "mysq": 4

The argument "mysq" could apply to the following Docker images, OpenShift image streams, or templates:

* Image stream "mysql" (tag "5.7") in project "openshift"
  Use --image-stream="openshift/mysql:5.7" to specify this image or template

* Template "mysql-persistent" in project "openshift"
  Use --template="openshift/mysql-persistent" to specify this image or template

* Template "cakephp-mysql-persistent" in project "openshift"
  Use --template="openshift/cakephp-mysql-persistent" to specify this image or template

* Template "dancer-mysql-persistent" in project "openshift"
  Use --template="openshift/dancer-mysql-persistent" to specify this image or template

```

When run from test/cmd/newapp.sh, it produces:

```
error: only a partial match was found for "mysq": "cmd-newapp/mysql:5.7"

The argument "mysq" only partially matched the following Docker image, OpenShift image stream, or template:

* Image stream "mysql" (tag "5.7") in project "cmd-newapp"
  Use --image-stream="cmd-newapp/mysql:5.7" to specify this image or template
```

Reply:
so do we have any test coverage for partial matches?